draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,39 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function extension involved in the expression
+must be supported.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
+
+Implementations MUST NOT support function extensions that are not registered.
+Note: if an implementation supports a function extension that is not registered,
+the implementation will not conform to this specification until the function extension is
+registered.
+To support a new function extension in an implementation, it is recommended that
+the implementation behaves by default in a "strict" mode that conforms to this specification and
+provides some way of behaving in a "non-strict" mode that supports the new function extension.
```

Comment:
How about:

> Custom functions MUST be supported only by a configuration option. Support of such functions MUST NOT be the default behavior.

Reply:
I'm trying to avoid constraining non-spec behaviour such as support for unregistered functions. Perhaps it isn't clear that the non-normative "Note:" is meant to extend to the end of the paragraph?

Reply:
> I'm trying to avoid constraining non-spec behaviour

But that's exactly what you're doing by saying "strict mode" and "non-strict mode."

Reply:
I've clarified that the strict vs non-strict stuff is a non-normative note. I think it's worth having even though it doesn't constrain non-spec behaviour at all.